Doesn't build on OSX

Trying to build this package locally on a Macbook and I get the following error:
```
error[E0432]: unresolved import `nix::sys::epoll::*`
  --> src/lib.rs:64:5
   |
64 | use nix::sys::epoll::*;
   |     ^^^^^^^^^^^^^^^^^^^ Could not find `epoll` in `nix::sys`
```

Version Info:
```
$ rustc --version
rustc 1.13.0

cargo 0.13.0 (eca9e15 2016-11-01)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,6 +6,10 @@ rust:
   - beta
   - nightly
 
+os:
+  - linux
+  - osx
+
 addons:
   apt:
     packages:
@@ -20,7 +24,7 @@ before_script:
 env:
   global:
     - TRAVIS_CARGO_NIGHTLY_FEATURE=""
-    - PATH=$HOME/.local/bin:$PATH
+    - PATH="$(python -m site --user-base)/bin:$PATH"
     - secure: "Ez5eIIgmsfNoV9NQ9H6KrAYX/Dxx6EUQEyyLMoSFvgJuMaPsyWlsDQDKbl9G+aNDNdSWYtYVFc4/fCW+H7a13HjVwB2CH2mChHSeguO8FSAkWns8tkZ+71SKL5/hg3Ig2m+kvfDjjV1O8seFEgTDP3enHFoRTmBSZHeNryo1ITM="
 
 script:
@@ -34,8 +38,8 @@ script:
     travis-cargo --only stable doc -- --features=tokio
 
 after_success:
-  - travis-cargo --only stable doc-upload
-  - travis-cargo coveralls --no-sudo --verify
+  - '[ $TRAVIS_OS_NAME == linux ] && travis-cargo --only stable doc-upload'
+  - '[ $TRAVIS_OS_NAME == linux ] && travis-cargo coveralls --no-sudo --verify'
 
 notifications:
   webhooks: http://rust-embedded-bot.herokuapp.com/travis

--- a/src/error.rs
+++ b/src/error.rs
@@ -10,6 +10,8 @@ pub enum Error {
     Unexpected(String),
     /// Invalid Path
     InvalidPath(String),
+    /// Operation not supported on target os
+    Unsupported(String),
 }
 
 impl ::std::error::Error for Error {
@@ -18,6 +20,7 @@ impl ::std::error::Error for Error {
             Error::Io(ref e) => e.description(),
             Error::Unexpected(_) => "An Unexpected Error Occurred",
             Error::InvalidPath(_) => "A Provided Path was invalid",
+            Error::Unsupported(_) => "Operation is not supported on target os",
         }
     }
 
@@ -35,6 +38,7 @@ impl fmt::Display for Error {
             Error::Io(ref e) => e.fmt(f),
             Error::Unexpected(ref s) => write!(f, "Unexpected: {}", s),
             Error::InvalidPath(ref s) => write!(f, "Invalid Path: {}", s),
+            Error::Unsupported(ref s) => write!(f, "Operation not supported on target os: {}", s),
         }
     }
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -61,6 +61,7 @@ use mio::Evented;
 #[cfg(feature = "mio-evented")]
 use mio::unix::EventedFd;
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
 use nix::sys::epoll::*;
 use nix::unistd::close;
 
@@ -469,6 +470,7 @@ impl PinPoller {
     }
 
     /// Create a new PinPoller for the provided pin number
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     pub fn new(pin_num: u64) -> Result<PinPoller> {
         let devfile: File = try!(File::open(&format!("/sys/class/gpio/gpio{}/value", pin_num)));
         let devfile_fd = devfile.as_raw_fd();
@@ -494,6 +496,11 @@ impl PinPoller {
         }
     }
 
+    #[cfg(not(any(target_os = "linux", target_os = "android")))]
+    pub fn new(pin_num: u64) -> Result<PinPoller> {
+        Err(Error::Unsupported("PinPoller".into()))
+    }
+
     /// Block until an interrupt occurs
     ///
     /// This call will block until an interrupt occurs.  The types
@@ -511,6 +518,7 @@ impl PinPoller {
     /// has occurred, but you could end up reading the same value multiple
     /// times as the value has changed back between when the interrupt
     /// occurred and the current time.
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     pub fn poll(&mut self, timeout_ms: isize) -> Result<Option<u8>> {
         try!(flush_input_from_file(&mut self.devfile, 255));
         let dummy_event = EpollEvent {
@@ -523,6 +531,11 @@ impl PinPoller {
             0 => None, // timeout
             _ => Some(try!(get_value_from_file(&mut self.devfile))),
         })
+    }
+
+    #[cfg(not(any(target_os = "linux", target_os = "android")))]
+    pub fn poll(&mut self, timeout_ms: isize) -> Result<Option<u8>> {
+        Err(Error::Unsupported("PinPoller".into()))
     }
 }
 

